xbmc/utils/FileUtils.cpp

```diff
@@ -226,3 +227,61 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   }
   return dateAdded;
 }
+
+bool CFileUtils::ZebraListAccessCheck(const std::string &filePath)
+{
+  // white/black list access checks, disallow exploits
+
+  // no access to these files,
+  // this can expose user/pass of remote servers
+  static const char *blacklist_files[] = {
+    "passwords.xml",
+    "sources.xml",
+    "guisettings.xml",
+    "advancedsettings.xml",
+    NULL
+  };
+
+  for (const char **ptr = blacklist_files; *ptr; ++ptr)
+  {
+    if (filePath.find(*ptr) != std::string::npos)
+    {
+      CLog::Log(LOGDEBUG,"http access denied");
+      return false;
+    }
+  }
+
+
+  char *fullpath;
+#if defined(TARGET_POSIX)
+  fullpath = realpath(filePath.c_str(), nullptr);
+#endif
+
+#if !defined(TARGET_POSIX)
```

Comment:
The `#endif` and the `#if !defined(TARGET_POSIX)` could be combined into an `#else`. But I'm not sure if `GetFullPathName()` is available on anything else but win32?